google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/v1/StreamingSubscriberConnection.java

```diff
@@ -148,12 +148,7 @@ public void onNext(StreamingPullResponse response) {
             public void run() {
               // Only request more if we're not shutdown.
               if (isAlive()) {
-                lock.lock();
-                try {
-                  thisRequestObserver.request(1);
-                } finally {
-                  lock.unlock();
-                }
+                thisRequestObserver.request(1);
```

Comment:
This can be called after onError has been called on thisRequestObserver. Are we certain that's valid?